### PR TITLE
Optimize perfect cube detection for large numbers with modular check

### PR DESCRIPTION
**Pull request by lighting9999**
1. Added modular arithmetic checks to quickly eliminate numbers that cannot be perfect cubes: · Last digit check: Only digits 0-9 can be cube endings · Modulo 7 check: Cubes can only be 0, 1, or 6 mod 7 · Modulo 9 check: Cubes can only be 0, 1, or 8 mod 9
2. Improved boundary estimation for very large numbers (n > 10^18): · Uses logarithmic approximation to calculate a reasonable upper bound · Reduces the binary search space significantly for extremely large values
3. Optimized computation in the binary search loop: · Added a check to avoid expensive cubic calculations for very large mid values · Uses integer division to prevent overflow in comparisons
4. Enhanced test coverage with comprehensive doctests: · Added tests for very large numbers (up to 10^300) · Included edge cases and error conditions · Maintained compatibility with existing tests

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by lighting9999** [comment]
@cclauss Can you help me review the PR?

File: maths/perfect_cube.py [hunk 1]
```diff
@@ -2,13 +2,42 @@ def perfect_cube(n: int) -> bool:
...
+    True
+    >>> perfect_cube(10**6 + 1)  # Large non-perfect cube
+    False
     """
+    # Handle negative numbers
+    if n < 0:
+        n = -n
+        is_negative = True
+    else:
+        is_negative = False
```
Comment (cclauss) [inline maths/perfect_cube.py:32]:
```suggestion
    if is_negative := n < 0:
        n = -n
```

**Comment by cclauss** [comment]
As discussed in CONTRIBUTING.md, do not add tests and code in the same pull request.  First, create a pull request that adds tests to the existing algorithm.  After that has landed, create a second pull request that changes the algorithm and modifies test results where the new algorithm generated different results than the old algorithm.

**Comment by lighting9999** [comment]
I'm so sorry. What should I do now?

**Review by lighting9999 — COMMENTED** [review]
I have just completed the suggested changes, what should I do now?
